tests/test-dirs/stack-or-heap.t/run.t

```diff
@@ -591,3 +599,35 @@ how to produce valid json.
   
   "could be stack or heap"
   
+  |  let z = Some x in
+  |               ^
+  
+  |  let z = Some x in
+  |          ^^^^^^
+  
+  "could be stack or heap"
+  
+  |  let z = Some x in
+  |               ^
+  
+  |  let z = Some x in
+  |          ^^^^^^
+  
+  "stack"
+  
+  |  let z = Some x in
+  |               ^
+  
+  |  let z = Some x in
+  |          ^^^^^^
+  
+  "could be stack or heap"
+  
+  |  let z = Some y in
+  |               ^
+  
+  |  let z = Some y in
+  |          ^^^^^^
+  
+  "heap"
+  
```

Comment:
I'd still like to see more tests of incomplete programs with "could be stack or heap". There's really only one test for this now (other than one that should print something else). It would be good to look for interactions, like `let f (local_ x) y = let z  = [x, Some y] in`. Maybe more? Maybe interactions between branches of a `match`, too.

Reply:
I added some more tests. I wanted to create a test where we know that a value is heap allocated but we get `could be heap or stack`, but I couldn't come up with one. I think this may be impossible (at least with the way this currently works)?

Reply:
Yes, that sounds right. It should be impossible, because we take ambiguous allocations and move them to `stack`. :)